### PR TITLE
[component] Small optimisation in BeamInterpolation

### PR DESCRIPTION
**Pull request by epernod**

### CODE REVIEW

**Review by alxbilger — APPROVED** [review]

File: src/BeamAdapter/component/BeamInterpolation.inl [hunk 1]
```diff
@@ -1210,18 +1210,19 @@ void BeamInterpolation<DataTypes>::InterpolateTransformUsingSpline(Transform& gl
...
+    Real invBx = 1-baryCoord;
+    Vec3NoInit posResult;
     Quat quatResult;
 
-    dP01 = P1-P0;
-    dP12 = P2-P1;
-    dP03 = P3-P0;
+    Vec3 dP01 = P1-P0;
+    Vec3 dP12 = P2-P1;
+    Vec3 dP03 = P3-P0;
```
Comment (fredroy) [inline src/BeamAdapter/component/BeamInterpolation.inl:1225]: cannot do suggestion 🤫
- const all "precomputed" stuff (bx, invbx, etc)
- there is now a Quat(noinit) as well

File: src/BeamAdapter/component/BeamInterpolation.inl [hunk 2]
```diff
@@ -1232,15 +1233,17 @@ void BeamInterpolation<DataTypes>::InterpolateTransformUsingSpline(Transform& gl
         quatResult = global_H_local0.getOrientation();
 
         //quatResult.slerp(global_H_local0.getOrientation(),global_H_local1.getOrientation(),bx,true);
-        posResult = P0 *(1-bx) + P3*bx;
+        posResult = P0 * invBx + P3*bx;
     }
     else
     {
+        Real bx2 = bx * bx;
+        Real invBx2 = invBx * invBx;
```
Comment (fredroy) [inline src/BeamAdapter/component/BeamInterpolation.inl:1241]:
```suggestion
        const Real bx2 = bx * bx;
        const Real invBx2 = invBx * invBx;

```

**Review by fredroy — CHANGES_REQUESTED** [review]

File: src/BeamAdapter/component/BeamInterpolation.inl (hunk 1 above)
Comment (epernod) [reply src/BeamAdapter/component/BeamInterpolation.inl:1225]: yes! 
In fact I added Quat(noinit) because of BA. but my changes has been lost somewhere.

Comment (epernod) [reply src/BeamAdapter/component/BeamInterpolation.inl]: ah.. using QuatNoInit make this plugin not anymore compatible with SOFA 22.06

**Review by fredroy — APPROVED** [review]

**Pull request merged**